Azure/sonic-buildimage
# [Services] Restart Platform-monitor service upon unexpected critical process exit.

- What I did
Restart Platform-monitor service if one of critical processes running in Platform-monitor container exited or crashed abnormally.
- How I did it
Generally I follow the framework created by Joe to implement this feature in Platform-monitor container.
**First**, add supervisor-proc-exit-listener event listener option in Supervisord configuration file in Platform-monitor docker container. Supervisord will read a list of critical processes for which to monitor the unexpected crashed and exited.
**Second**, configure Platform-monitor.service to always auto-restart the service if it stops, with a delay of 30 seconds. Also set a rate limit of 3 restarts within 20 minutes (1200 seconds).
- How to verify it
On your switch device, please use docker _ps_ command to list all running docker containers.
Then use _docker exec -it container_id bash_ to login target container. Typing _top_ command
on the shell will display all the processes dynamically and you will spot the process id of one
of the critical processes. Finally type the command _kill -9 process_id_ to terminate one process.
After exiting the container, you can use _watch -n 1 docker ps_ to dynamically see the restart
of Platform-monitor container.
